src/inputs/DateFields/DateFieldBase.test.tsx

```diff
@@ -34,6 +34,24 @@ describe(DateFieldBase, () => {
     expect(r.date_helperText().textContent).toBe("Helper text");
   });
 
+  it("renders correctly without calendar icon button", async () => {
+    // Given a DateField without a calendar icon
+    const r = await render(
+      <DateFieldBase
+        mode="single"
+        value={undefined}
+        label="Date"
+        onChange={noop}
+        placeholder="Select a date"
+        hideCalendarIcon
+      />,
+    );
+    // Calendar icon isn't rendered
```

Comment:
Nit suggestion: I really like the inline comments here, so +1; even better would be to double down on the Given/When/Then prefixes from BDD and do "// Then the calendar icon isn't render" and "// And the placeholder is still there".